lib/api.common.js

```diff
@@ -612,23 +610,25 @@ api.common.logApiMethod = (
     }
     const logArgs = api.json.stringify(args);
     if (impress && impress.log) {
-      impress.log.debug(
-        fnPath + '(' + logArgs.substring(1, logArgs.length - 1) + ', callback)'
-      );
+      const fPar = logArgs.substring(1, logArgs.length - 1);
+      const fMsg = fnPath + '(' + fPar + ', callback)';
+      impress.log.debug(fMsg);
       const stack = new Error().stack.toString().split('\n');
       impress.log.debug(stack[2].trim());
     }
-    if (callback) args.push(() => {
-      const args = [];
-      Array.prototype.push.apply(args, args);
-      const logArgs = api.json.stringify(args);
-      if (impress && impress.log) {
-        impress.log.debug(
-          fnPath + ' callback(' + logArgs.substring(1, logArgs.length - 1) + ')'
-        );
-      }
-      callback(...args);
-    });
+    if (callback) {
+      args.push(() => {
+        const args = [];
+        Array.prototype.push.apply(args, args);
```

Comment:
Does nothing. I suppose it is a reminiscent of times when the second argument was, as far as I understand, `arguments`. If that's the case, lines 621 and 622 must be removed because not only 622 is useless, 621 introduces a bug shadowing `args` from the outer scope and forcing all the logged method arguments to be empty.